crates/flox/src/commands/channel.rs

```diff
@@ -76,122 +80,118 @@ pub struct Search {
-                    }
-
-                    if let Some(desc) = entry.get("description").unwrap().as_str() {
-                        print!(" - {}", desc)
-                    }
-                    println!();
-                }
-            }
+        let (results, exit_status) = do_search(&search_params_json)?;
+
+        // Render what we have no matter what, then indicate whether we encountered an error.
+        // FIXME: We may have warnings on `stderr` even with a successful call to `pkgdb`.
+        //        We aren't checking that at all at the moment because better overall error handling
+        //        is coming in a later PR.
+        if self.json {
+            render_search_results_json(&results)?;
+        } else {
+            render_search_results(&results)?;
+        }
+        if exit_status.success() {
             Ok(())
         } else {
             bail!(
-                "pkgdb exited with status code {}",
-                output.status.code().unwrap_or(-1)
+                "pkgdb exited with status code: {}",
+                exit_status.code().unwrap_or(-1),
             );
         }
     }
 }
 
+fn construct_search_params(search_term: &str, flox: &Flox) -> Result<SearchParams> {
+    let channels = flox
+        .channels
+        .iter()
+        .filter_map(|entry| {
+            if HIDDEN_CHANNELS.contains_key(&*entry.from.id) {
+                None
+            } else if DEFAULT_CHANNELS.contains_key(&*entry.from.id) {
+                Some((ChannelType::Flox, entry))
+            } else {
+                Some((ChannelType::User, entry))
+            }
+        })
+        .sorted();
+
+    // Create `registry` parameter for `pkgdb`
+    let mut priority: Vec<String> = Vec::new();
+    let mut inputs = HashMap::new();
+    for (_, entry) in channels {
+        priority.push(entry.from.id.to_string());
+        let input = RegistryInput {
+            from: entry.to.clone(),
+            // TODO: handle `subtrees` and `stabilities`
+            subtrees: None,
+            stabilities: None,
+        };
+        inputs.insert(entry.from.id.to_string(), input);
+    }
+    let registry = Registry {
+        inputs,
+        priority,
+        defaults: RegistryDefaults::default(),
+    };
+
+    // We've already checked that the search term is Some(_)
+    let query = Query::from_str(search_term)?;
+
+    Ok(SearchParams {
+        registry,
+        query,
+        ..SearchParams::default()
+    })
+}
+
+// This is likely to change significantly after the output format of search results is specced out
+fn render_search_results(search_results: &SearchResults) -> Result<()> {
+    let summarized_results = search_results
+        .results
+        .iter()
+        .map(|r| {
+            let path_components = r.attr_path.clone();
+            let flake_attr = if !DEFAULT_CHANNELS.contains_key(r.input.as_str()) {
+                vec![r.input.clone(), path_components.join_with(".").to_string()]
```

Comment:
Why not just use `join`?

Reply:
Ah, this is a leftover from an optimization I had in at one point. I'll take it out in the next PR, the ~30min CI runs are killing me.

Reply:
Fixed